tests/lib/rules/max-classes-per-file.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Tests for max-classes-per-file rule.
+ * @author James Garbutt <https://github.com/43081j>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/max-classes-per-file"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("max-classes-per-file", rule, {
+    valid: [
+        {
+            code: "class Foo {}",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var x = class {};",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var x = 5;",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "class Foo {}\nvar cls = class {};",
+            options: [1, { excludeClassExpressions: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "class Foo {}\nclass Bar {}",
+            options: [2],
+            parserOptions: { ecmaVersion: 6 }
+        }
```

Comment:
It would be good to have some examples with `options: [1]` explicitly, just to protect against regressions.